solidity/test/bridge/Bridge.MovingFunds.test.ts

```diff
@@ -3075,6 +3170,466 @@ describe("Bridge - Moving funds", () => {
+            const testData: {
+              testName: string
+              walletState: number
+              additionalSetup?: () => Promise<void>
+              additionalAssertions?: () => Promise<void>
+            }[] = [
+              {
+                testName:
+                  "when the wallet is in the Live state but the wallet is not the active one",
+                walletState: walletState.Live,
+                additionalSetup: async () => {
+                  // The active wallet is a different wallet than the tested one
+                  await bridge.setActiveWallet(
+                    "0x0b9f85c224b0e018a5865392927b3f9e16cf5e79"
+                  )
+                },
+                additionalAssertions: async () => {
+                  it("should decrease the live wallets count", async () => {
+                    expect(await bridge.liveWalletsCount()).to.be.equal(0)
+                  })
+
+                  it("should not unset the active wallet", async () => {
+                    expect(
+                      await bridge.activeWalletPubKeyHash()
+                    ).to.be.not.equal(
+                      "0x0000000000000000000000000000000000000000"
+                    )
+                  })
+                },
+              },
+              {
+                testName:
+                  "when the wallet is in the Live state and the wallet is the active one",
+                walletState: walletState.Live,
+                additionalSetup: async () => {
+                  await bridge.setActiveWallet(
+                    movedFundsSweepRequest.walletPubKeyHash
+                  )
+                },
+                additionalAssertions: async () => {
+                  it("should decrease the live wallets count", async () => {
+                    expect(await bridge.liveWalletsCount()).to.be.equal(0)
+                  })
+
+                  it("should unset the active wallet", async () => {
+                    expect(await bridge.activeWalletPubKeyHash()).to.be.equal(
+                      "0x0000000000000000000000000000000000000000"
+                    )
+                  })
+                },
+              },
+              {
+                testName: "when the wallet is in the MovingFunds state",
+                walletState: walletState.MovingFunds,
+                additionalSetup: async () => {},
+                additionalAssertions: async () => {},
+              },
+            ]
+
+            testData.forEach((test) => {
+              context(test.testName, async () => {
+                let tx: ContractTransaction
+
+                before(async () => {
+                  await createSnapshot()
+
+                  await bridge.setWallet(
+                    movedFundsSweepRequest.walletPubKeyHash,
+                    {
+                      ...(await bridge.wallets(
+                        movedFundsSweepRequest.walletPubKeyHash
+                      )),
```

Comment:
I do not entirely follow what is happening here. Can you clarify please?

Reply:
We change the wallet `state` while preserving other fields. We don't have an update function in the stub so we just use the getter to get wallet fields and set them through the setter with `state` field overwritten.

Reply:
Ah! Yeah, it makes sense. Can you please copy&paste this explanation to the comment above this line?

Reply:
(in the next PR, I am merging this one)